### PR TITLE
Create dist Target, separate makefile

### PR DESCRIPTION
**Pull request by micbar**
## Description
Create Makefile.release, remove all dist stuff from Makefile.

## Motivation and Context
Separate the Dev and Build Process

## How Has This Been Tested?
Tested manually

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: Makefile.release [hunk 1]
```diff
@@ -0,0 +1,88 @@
+SHELL := /bin/bash
+
+COMPOSER_BIN := $(shell command -v composer 2> /dev/null)
+ifndef COMPOSER_BIN
+    $(error composer is not available on your system, please install composer)
+endif
+YARN := $(shell command -v yarn 2> /dev/null)
```
Comment (patrickjahns) [inline Makefile.release]: I didn't see yarn being used in the makefile - can you quickly share why you want to keep the check in the makefile?

Comment (micbar) [reply Makefile.release]: Ok, we do not need that.

**Comment by IljaN** [comment]
Rebased with master

File: Makefile.release (hunk 1 above)
Comment (micbar) [reply Makefile.release]: done

**Pull request merged**